solr/core/src/java/org/apache/solr/handler/loader/JsonLoader.java

```diff
@@ -249,14 +251,27 @@ public void handle(Map<String, Object> record, String path) {
     private SolrInputDocument buildDoc(Map<String, Object> m) {
       SolrInputDocument result = new SolrInputDocument();
       for (Map.Entry<String, Object> e : m.entrySet()) {
-        if (e.getKey() == null) {// special case. JsonRecordReader emits child docs with null key
+        if (entryIsChildDoc(e.getValue())) {// special case. JsonRecordReader emits child docs with null key
           if (e.getValue() instanceof List) {
             List value = (List) e.getValue();
             for (Object o : value) {
-              if (o instanceof Map) result.addChildDocument(buildDoc((Map) o));
+              if (o instanceof Map) {
+                if (anonChildDocFlag) {
+                  result.addChildDocument(buildDoc((Map) o));
+                } else {
+                  if(!result.containsKey(e.getKey())) {
+                    result.setField(e.getKey(), new ArrayList<>(1));
```

Comment:
This looks unnecessary; SolrInputDocument internally handles ensuring multiple values can be captures.

Reply:
It can handle it internally, but we have to ensure a JSON list is added as a list of one document, and not as a single document, to remain true to the original JSON document.

Reply:
ok so you are trying to retain the semantic information that the input was an array (potentially multi-valued) even if the particular input doc only had one value.  Maybe say more or that, hinting at why we even care.  (why do we care)?  For regular field values, we don't at this level -- the schema holds multiValue info so it's dealth with later.  Again I'm liking my suggestion of putting a virtual child doc field in the schema as it addresses the desire to know this semantic info, plus I think it may add some consistency (avoids special case you're doing here)

Reply:
if we wanted to build the original JSON document as it was indexed, we would have to store it as a single valued array if that was the case. I'm afraid I did not quite get what you meant by using virtual fields, what for?

Reply:
By "virtual field" I'm referring to my earlier idea about having a special field in the schema that designates a child document with this relation.  it would have a name, multiValued option, and probably that's it.  https://issues.apache.org/jira/browse/SOLR-12298?focusedCommentId=16467523&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16467523

Reply:
Any way; lets not increase the scope of this particular issue with my idea to add this info to the schema. It seems adequate to force the use of an internal array for a single value in SolrInputField that we know originated from an array.